common/common-client/src/main/java/org/ow2/proactive/authentication/FileLoginModule.java

```diff
@@ -179,19 +186,34 @@ public boolean login() throws LoginException {
      * check group membership from group file.
      * @param username user's login
      * @param password user's password
-     * @param printErrorMessage if a message should be printed if the password is incorrect.
+     * @param isNotFallbackAuthentication true if this method is not called inside a fallback mechanism
      * @return true user login and password are correct, and requested group is authorized for the user
      * @throws LoginException if authentication or group membership fails.
      */
-    protected boolean logUser(String username, String password, boolean printErrorMessage) throws LoginException {
+    protected boolean logUser(String username, String password, boolean isNotFallbackAuthentication)
+            throws LoginException {
+
+        if (isNotFallbackAuthentication) {
+            removeOldFailedAttempts(username);
+
+            if (tooManyFailedAttempts(username)) {
+                String message = "Too many failed login/attempts, please try again in " +
+                                 retryInHowManyMinutes(username) + " minutes.";
+                logger.warn("[" + FileLoginModule.class.getSimpleName() + "] " + message);
+                throw new FailedLoginException(message);
+            }
+        }
```

Comment:
What happens if the fallback mechanism is used? the user can try to login without any limitation?

Reply:
in the fallback mechanism the file login module is checked first then the main module is checked next (ldap or pam).

I disabled the maximum failed attempts on the file login module in that case (but it is kept when both login attempts fail), otherwise accounts would be blocked all the time for non file-based users.

Reply:
Fair enough :)